### ISSUE TITLE
cmd/compile: should use FNEG for float32/64 on ppc64

### ISSUE THREAD

**Issue by minux** [golang/go#9598]
It always generates FMUL -1 for OMINUS now.
However, Power does support double precision FNEG, it just
doesn't support single precision FNEGS.

But it's still fine to use FNEG for both float32 and float64, as
Power ISA documents that FNEG simply inverts bit 0.

**Comment by rsc** [comment]
I believe this is too unimportant for right now. Finishing the port (especially those nagging cgo details) takes priority over microoptimizations. Removing milestone.

**Comment by josharian** [comment]
@dr2chase does ppc64 SSA do this? Can we close this issue?

**Comment by dr2chase** [comment]
It should be fixed in 1.8:

```
(Neg64F x) -> (FNEG x)
(Neg32F x) -> (FNEG x)
```
```
    {name: "FNEG", argLength: 1, reg: fp11, asm: "FNEG"},   // -arg0 (floating point)
```


```
	case ssa.OpPPC64NEG, ssa.OpPPC64FNEG, ssa.OpPPC64FSQRT, ssa.OpPPC64FSQRTS, ssa.OpPPC64FCTIDZ, ssa.OpPPC64FCTIWZ, ssa.OpPPC64FCFID, ssa.OpPPC64FRSP:
		r := gc.SSARegNum(v)
		p := gc.Prog(v.Op.Asm())
```

**Comment by josharian** [comment]
Thanks. Closing.
